ttnn/cpp/ttnn/operations/embedding/device/kernels/dataflow/embedding_ind_tilized.cpp

```diff
@@ -0,0 +1,220 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "dataflow_api.h"
+#include "debug/dprint.h"
+
+void kernel_main() {
+    const std::uint32_t input_dram_buffer_src_addr = get_arg_val<uint32_t>(0);
+    const std::uint32_t weights_dram_buffer_src_addr = get_arg_val<uint32_t>(1);
+    const std::uint32_t tile_offset = get_arg_val<uint32_t>(2);
+    const std::uint32_t face_offset = get_arg_val<uint32_t>(3);
+    const std::uint32_t num_rows = get_arg_val<uint32_t>(4);
+
+    const std::uint32_t curr_col = get_arg_val<uint32_t>(5);
+    const std::uint32_t starting_index = get_arg_val<uint32_t>(6);
+
+#define in_is_dram get_compile_time_arg_val(0) == 1
+#define in_stick_size_is_power_of_two get_compile_time_arg_val(1) == 1
+    constexpr uint32_t input_page_size = get_compile_time_arg_val(2);
+#if (in_stick_size_is_power_of_two)
+    constexpr uint32_t log_base_2_of_input_page_size = get_compile_time_arg_val(3);
+    const InterleavedPow2AddrGen<in_is_dram> input = {
+        .bank_base_address = input_dram_buffer_src_addr,
+        .log_base_2_of_page_size = log_base_2_of_input_page_size  // TODO(AP): refactor
+    };
+#else
+    const InterleavedAddrGen<in_is_dram> input = {
+        .bank_base_address = input_dram_buffer_src_addr, .page_size = input_page_size};
+#endif
+
+#define weights_is_dram get_compile_time_arg_val(4) == 1
+#define weight_stick_size_is_power_of_two get_compile_time_arg_val(5) == 1
+    constexpr uint32_t weight_stick_size = get_compile_time_arg_val(6);
+#if (weight_stick_size_is_power_of_two)
+    constexpr uint32_t log_base_2_of_weights_page_size = get_compile_time_arg_val(7);
+    const InterleavedPow2AddrGen<weights_is_dram> weights = {
+        .bank_base_address = weights_dram_buffer_src_addr,
+        .log_base_2_of_page_size = log_base_2_of_weights_page_size  // TODO(AP): refactor
+    };
+#else
+    const InterleavedAddrGen<weights_is_dram> weights = {
+        .bank_base_address = weights_dram_buffer_src_addr, .page_size = weight_stick_size};
+#endif
+
+    constexpr uint32_t row_length = get_compile_time_arg_val(8);
+    constexpr uint32_t input_block_size_bytes = get_compile_time_arg_val(9);
+
+    constexpr uint32_t cb_id_in0 = 0;
+    constexpr uint32_t cb_id_in1 = 1;
+    constexpr uint32_t cb_id_in2 = 2;
+
+    constexpr uint32_t face_size = 16;
+    constexpr uint32_t tile_height = 32;
```

Comment:
Can we send this in as a param from host using tile shape. Look up how to get the tile height from the tile shape from the tensor.

Tldr, in the future we are going to support tiles that are not 32x32, so we want to get that from the tile shape api.

Reply:
When I say this, I mean the tile height

Reply:
Also, we are moving towards passing in cb id as compile-time args instead of hard-coding.

Reply:
removing hardcoded cb ids is related to the cleanup requested here https://github.com/tenstorrent/tt-metal/pull/16296#issuecomment-2560374212